ReactNativeClient/lib/services/searchengine/SearchEngine.js

```diff
@@ -474,18 +477,33 @@ class SearchEngine {
 			const fuzzyTextTerms = mergedFuzzyText.map(x => { return { name: 'text', value: x.word, negated: false, score: x.score }; });
 			const fuzzyTitleTerms = mergedFuzzyTitle.map(x => { return { name: 'title', value: x.word, negated: false, score: x.score }; });
 			const fuzzyBodyTerms = mergedFuzzyBody.map(x => { return { name: 'body', value: x.word, negated: false, score: x.score }; });
-			const phraseTextTerms = phraseSearches.map(x => { return { name: 'text', value: x, negated: false, score: 0 }; });
 
+			// Remove previous text, title and body and replace with fuzzy versions
 			allTerms = allTerms.filter(x => (x.name !== 'text' && x.name !== 'title' && x.name !== 'body'));
 
-			allFuzzyTerms = allTerms.concat(fuzzyTextTerms).concat(fuzzyTitleTerms).concat(fuzzyBodyTerms).concat(phraseTextTerms);
+			// The order matters here!
+			// The text goes first, then title, then body, then phrase and finally wildcard
+			// This is because it needs to match with numFuzzyMathches.
+			allFuzzyTerms = allTerms.concat(fuzzyTextTerms).concat(fuzzyTitleTerms).concat(fuzzyBodyTerms).concat(phraseTextSearch).concat(wildCardSearch);
 
 			const allTextTerms = allFuzzyTerms.filter(x => x.name === 'title' || x.name === 'body' || x.name === 'text');
 			for (let i = 0; i < allTextTerms.length; i++) {
+				// Phrase searches and wildcard searches will get a fuzziness score of zero.
+				// This means that they will go first in the sort order (Even if there are other words with matches in the title)
+				// Undesirable?
 				fuzzyScore.push(allFuzzyTerms[i].score ? allFuzzyTerms[i].score : 0);
 			}
```

Comment:
Any suggestions on what to do for their fuzzy scores?

Reply:
Leaving it as it is seems ok, as I expect users will rarely use complex combinations of filters, wildcards, etc and they do it might just be one edge case we don't support. Or can you think if a specific case where it could be a problem? If so perhaps add it as a comment so next time we know where to look.

Reply:
@naviji, please let me know what you'd prefer to do about this. We just need this pull request before doing release 1.1.

Reply:
I guess the most important case would be in GoToAnything where everything is a wildcard search.
I'll add it as a comment. We'll make fuzzy search work with GoToAnything in another release.

PS. Sorry for the delay. College has opened and it's been a busy week.

Reply:
Right, that makes sense, it would indeed affects GotoAnything although this is fine for now. I'll add a small comment as a reminder.